.github/scripts/all_tests.sh

```diff
@@ -3,6 +3,35 @@
 # exit if error
 set -e
 
+
+# check that in scripts directory
+
+CURRENT_DIR="$(pwd)"
+SCRIPTS_DIR='obsidian-hub/.github/scripts'
+
+if ! [[ "$CURRENT_DIR" =~ "$SCRIPTS_DIR" ]] ; then
+    echo "Your current directory is $CURRENT_DIR"
+    echo "It needs to be ...$SCRIPTS_DIR"
+    exit 1
+fi
+
+
+# check if virtual environment is activated
+
+PYTHON="$(which python)"
+PARTIAL_PATH='venv/bin/python'
+
+if ! [[ "$PYTHON" =~ "$PARTIAL_PATH" ]] ; then
```

Comment:
Does this require that people's virtual environment is called venv?
Does it enforce a particular location?
Does it enforce use of a virtual environment? What if they had applied the requirements.txt to some other python, for a working environment?
What if they had done the activate, but not applied the requirements?

Checks are hard!

Reply:
It is meant to enforce using a virtual environment. I didn't think about the requirements.txt because that's only a one time set-up. I think it's not worth being automated.

Reply:
The partial path is the end of the result of `which python` after sourcing the venv activation script.
The condition checks whether it doesn't match partly.